src/bitmessagekivy/baseclass/draft.py

```diff
@@ -0,0 +1,203 @@
+# pylint: disable=unused-argument, import-error, too-many-arguments
+# pylint: disable=unnecessary-comprehension, no-member, no-name-in-module
+
+"""
+draft.py
+==============
+
+Draft screen
+
+"""
+
+from functools import partial
+import time
+import os
+
+from kivy.clock import Clock
+from kivy.properties import (
+    ListProperty,
+    StringProperty
+)
+from kivy.uix.screenmanager import Screen
+
+import state
+
+from bitmessagekivy import kivy_helper_search
+from bitmessagekivy.baseclass.common import (
+    showLimitedCnt, toast, ThemeClsColor,
+    SwipeToDeleteItem, ShowTimeHistoy
+)
+from bitmessagekivy.baseclass.maildetail import MailDetail
+from bitmessagekivy.baseclass.draft_widgets import HelperDraft
+
+from addresses import decodeAddress
+from bmconfigparser import config
+
+from helper_sql import sqlExecute
+
+
+class Draft(Screen, HelperDraft):
+    """Draft screen class for kivy Ui"""
+
+    data = ListProperty()
+    account = StringProperty()
+    queryreturn = ListProperty()
+    has_refreshed = True
+
+    def __init__(self, *args, **kwargs):
+        """Method used for storing draft messages"""
+        super(Draft, self).__init__(*args, **kwargs)
+        if state.association == '':
+            if state.kivyapp.variable_1:
+                state.association = state.kivyapp.variable_1[0]
+        Clock.schedule_once(self.init_ui, 0)
+
+    def init_ui(self, dt=0):
+        """Clock Schdule for method draft accounts"""
+        self.sentaccounts()
+        print(dt)
+
+    def sentaccounts(self):
+        """Load draft accounts"""
+        self.loadDraft()
+
+    def loadDraft(self, where="", what=""):
+        """Load draft list for Draft messages"""
+        self.account = state.association
+        xAddress = 'fromaddress'
+        self.ids.tag_label.text = ''
+        self.draftDataQuery(xAddress, where, what)
+        if self.queryreturn:
+            self.ids.tag_label.text = 'Draft'
+            self.set_draftCnt(state.draft_count)
+            self.set_mdList()
+            self.ids.scroll_y.bind(scroll_y=self.check_scroll_y)
+        else:
+            self.set_draftCnt('0')
+            self.ids.ml.add_widget(self.default_label_when_empty())
+
+    def draftDataQuery(self, xAddress, where, what, start_indx=0, end_indx=20):
+        """This methosd is for retrieving draft messages"""
+        self.queryreturn = kivy_helper_search.search_sql(
+            xAddress, self.account, "draft", where, what,
+            False, start_indx, end_indx)
+
+    @staticmethod
+    def set_draftCnt(Count):
+        """This method set the count of draft mails"""
+        draftCnt_obj = state.kivyapp.root.ids.content_drawer.ids.draft_cnt
+        draftCnt_obj.ids.badge_txt.text = showLimitedCnt(int(Count))
+
+    def set_mdList(self):
```

Comment:
maybe this can be also generalised into a parent class, with `sent` and `draft` being child classes